### ISSUE TITLE
Make shell understand non-interactive mode

### ISSUE THREAD

**Issue by jvalkeal** [spring-projects/spring-shell#187]
In skipper we wanted to run commands as non-interactive mode
```
java -jar spring-cloud-skipper-shell/target/spring-cloud-skipper-shell-1.0.0.BUILD-SNAPSHOT.jar search

vs.

java -jar spring-cloud-skipper-shell/target/spring-cloud-skipper-shell-1.0.0.BUILD-SNAPSHOT.jar
skipper:>search
```
We found a quick hack to this in [InteractiveModeApplicationRunner](https://github.com/spring-cloud/spring-cloud-skipper/blob/master/spring-cloud-skipper-shell-commands/src/main/java/org/springframework/cloud/skipper/shell/command/support/InteractiveModeApplicationRunner.java#L99).

What we also wanted to do is to return non zero from an non-interactive mode when underlying shell command fails. This turned out to be impossible, afaik, to implement atop of shell core in a skipper. There is `ExitRequestResultHandler` and `ExitRequest` but I wasn't able to use those without breaking interactive mode. I thought by customising shell, created #182 but that was also dead end.

I think we need to make shell core to better understand non-interactive run.

**Comment by ericbottard** [comment]
One option may be to allow override of `ResultHandlers` (`ThrowableResultHandler` in that case)

**Comment by Stryter** [comment]
So, I've been trying to get Spring Shell to execute in a non-interactive way and I haven't had any luck, so I went in search to see if this was even possible with Spring Shell. I came across this thread, and from what I am reading it looks like this is not possible, yet.

Am I correct in my understanding? If so, will this explicit functionality be coming in the future?

For the sake of clarity, what I am trying to accomplish is something like the following:

`java -jar utility.jar fancy-command --option1 arg`

And have spring shell execute the command and immediately exit.

Thanks in advance for your input!
